lib/pedicel/ec.rb

```diff
@@ -6,10 +6,15 @@ def ephemeral_public_key
       Base64.decode64(@token['header']['ephemeralPublicKey'])
     end
 
+    # Decrypt self.encrypted_data() finding the key from either
+    # symmetric_key, og
+    #
     def decrypt(symmetric_key: nil, merchant_id: nil, certificate: nil, private_key: nil,
-                ca_certificate_pem: @config[:apple_root_ca_g3_cert_pem], now: Time.now)
-      raise ArgumentError 'invalid argument combination' unless \
-        !symmetric_key.nil? ^ ((!merchant_id.nil? ^ !certificate.nil?) && !private_key.nil?)
+                ca_certificate_pem: @config[:trusted_ca_pem], now: Time.now)
+      raise Pedicel::ArgumentError, 'invalid argument combination' unless \
+        !symmetric_key.nil? ^
+        ((!merchant_id.nil? ^ !certificate.nil?) && private_key)
```

Comment:
This breaks the alignment of the comments below.

Do not fear a few long lines 🙂

Reply:
I can live with the long lines then :)
Though this break was more for clarity, with which it didn't help, of course. Going to look at it again.

Reply:
At first sight, I did only notice the wrapping. Now I notice `!private_key.nil?` -> `private_key`.

I don't know if I like it. Generally I certainly do because `if foo` reads better than `if !foo.nil?`, but since `!x.nil?` is necessary earlier on the line (because `^` will not coerce), I don't really know.

I'm indifferent and I guess you prefer the coercing version, let's go with it 🙂